File: android/walletsdk/src/main/java/leaf/prod/walletsdk/model/news/IndexType.java
/**
 * Created with IntelliJ IDEA.
 * User: kenshin wangchen@loopring.org
 * Time: 2018-12-24 4:10 PM
 * Cooperation: loopring.org 路印协议基金会
 */
package leaf.prod.walletsdk.model.news;

public enum IndexType {

    BULLINDEX("bull_index"),

    BEARINDEX("bear_index"),

    FORWARDNUM("forward_num"),

    READNUM("read_num"),

    UNKNOWN("unknown");

    private String description;

    IndexType(String description) {
        this.description = description;
    }

    public String getDescription() {
        return description;
    }
}

File: loopr-ios/Models/News/IndexType.swift
//
//  OrderType.swift
//  loopr-ios
//
//  Created by kenshin on 2018/5/28.
//  Copyright © 2018年 Loopring. All rights reserved.
//

import Foundation

enum IndexType: String, CustomStringConvertible {
    
    case bullIndex = "bull_index"
    case bearIndex = "bear_index"
    case forwardNum = "forward_num"
    case unknown = "unknown"
    
    var description: String {
        switch self {
        case .bullIndex: return "bull_index"
        case .bearIndex: return "bear_index"
        case .forwardNum: return "forward_num"
        case .unknown: return "unknown"
        }
    }
}
